common/nexus/failure.go

```diff
@@ -31,14 +32,17 @@ import (
 	failurepb "go.temporal.io/api/failure/v1"
 	nexuspb "go.temporal.io/api/nexus/v1"
 	"go.temporal.io/api/serviceerror"
+	"go.temporal.io/server/common"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/encoding/protojson"
 )
 
+var failureTypeString = string((&failurepb.Failure{}).ProtoReflect().Descriptor().FullName())
+
 // ProtoFailureToNexusFailure converts a proto Nexus Failure to a Nexus SDK Failure.
-// Always returns a non-nil value.
-func ProtoFailureToNexusFailure(failure *nexuspb.Failure) *nexus.Failure {
-	return &nexus.Failure{
+func ProtoFailureToNexusFailure(failure *nexuspb.Failure) nexus.Failure {
```

Comment:
If `failure` is nil, this will return a object with empty fields. Is this the desired behavior?

Reply:
Yeah, it's desired.